src/features/titles/with-test-data.js

```diff
@@ -0,0 +1,118 @@
+/* global wp */
+import deepmerge from 'deepmerge';
+
+const { apiFetch } = wp;
+const { withSelect, withDispatch } = wp.data;
+const { compose, withState } = wp.compose;
+const { __ } = wp.i18n;
+
+const withTestData = compose(
+	withState( {
+		isSaving: false,
+		prevTitles: [],
+		error: false,
+	} ),
+	withSelect( select => {
+		const ab_tests = select( 'core/editor' ).getEditedPostAttribute( 'ab_tests' );
+		const test = ab_tests.titles;
+		const post = select( 'core/editor' ).getCurrentPost();
+		const originalTitles = select( 'core/editor' ).getCurrentPostAttribute( 'ab_test_titles' ) || [];
+		const title = select( 'core/editor' ).getEditedPostAttribute( 'title' ) || '';
+		const titles = select( 'core/editor' ).getEditedPostAttribute( 'ab_test_titles' ) || [];
+		const currentPostType = select( 'core/editor' ).getCurrentPostType();
+		const postType = select( 'core' ).getPostType( currentPostType );
+
+		return {
+			ab_tests,
+			originalTitles,
+			post,
+			postType,
+			test,
+			title,
+			titles,
+		};
+	} ),
+	withDispatch( ( dispatch, props ) => {
+		const { ab_tests, post, postType, setState } = props;
+
+		const saveTest = async data => {
+			setState( { isSaving: true } );
+
+			try {
+				await apiFetch( {
+					path: `/wp/v2/${ postType.rest_base }/${ post.id }` +
+						`?context=edit&_timestamp=${ Date.now() }`,
+					method: 'PATCH',
+					body: JSON.stringify( data ),
+				} );
+			} catch ( error ) {
+				setState( { error: error } );
+			}
+
+			setState( { isSaving: false } );
+		}
+
+		const updateTest = async ( test = {}, titles = false, save = false ) => {
+			const data = deepmerge( {
+				ab_tests,
+			}, {
```

Comment:
Can this do `const data = { ab_tests: deepmerge( ab_tests, { titles: test } ) }` instead? No need to merge two static objects at the higher level